public/app/plugins/datasource/cloudwatch/components/ConfigEditor.tsx

```diff
@@ -7,26 +8,75 @@ import {
   DataSourcePluginOptionsEditorProps,
   onUpdateDatasourceJsonDataOption,
   updateDatasourcePluginJsonDataOption,
+  SelectableValue,
+  updateDatasourcePluginOption,
 } from '@grafana/data';
-import { Input, InlineField } from '@grafana/ui';
+import { getBackendSrv } from '@grafana/runtime';
+import { Input, InlineField, MultiSelect } from '@grafana/ui';
 import { notifyApp } from 'app/core/actions';
-import { createWarningNotification } from 'app/core/copy/appNotification';
+import { createErrorNotification, createWarningNotification } from 'app/core/copy/appNotification';
 import { getDatasourceSrv } from 'app/features/plugins/datasource_srv';
-import { store } from 'app/store/store';
+import { dispatch, store } from 'app/store/store';
 
 import { CloudWatchDatasource } from '../datasource';
 import { CloudWatchJsonData, CloudWatchSecureJsonData } from '../types';
+import { toOption } from '../utils/utils';
 
+import { MAX_LOG_GROUPS, MAX_VISIBLE_LOG_GROUPS } from './LogsQueryField';
 import { XrayLinkConfig } from './XrayLinkConfig';
 
 export type Props = DataSourcePluginOptionsEditorProps<CloudWatchJsonData, CloudWatchSecureJsonData>;
 
 export const ConfigEditor: FC<Props> = (props: Props) => {
   const { options } = props;
+  const { defaultLogGroups, logsTimeout } = options.jsonData;
 
-  const datasource = useDatasource(options.name);
+  const { datasource, setSaved } = useDatasource(options.name);
   useAuthenticationWarning(options.jsonData);
-  const logsTimeoutError = useTimoutValidation(props.options.jsonData.logsTimeout);
+  const logsTimeoutError = useTimoutValidation(logsTimeout);
+  const [logGroups, setLogGroups] = useState<SelectableValue[]>([]);
+  const [loadingLogGroups, setLoadingLogGroups] = useState(false);
```

Comment:
If you wrap the `loadLogGroups` in a `useAsyncFunc`, the same way it's done [here](https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/cloudwatch/components/Dimensions/FilterItem.tsx#L64), you should be able to get rid of the two states in this and the previous line.

Reply:
code removed